src/jit/flowgraph.cpp

```diff
@@ -5726,33 +5749,14 @@ void Compiler::fgFindBasicBlocks()
 
     /* Now create the basic blocks */
 
-    fgMakeBasicBlocks(info.compCode, info.compILCodeSize, jumpTarget);
+    unsigned retBlocks = fgMakeBasicBlocks(info.compCode, info.compILCodeSize, jumpTarget);
 
     if (compIsForInlining())
     {
-        if (compInlineResult->IsFailure())
-        {
-            return;
-        }
-
```

Comment:
Its checked after setting the no return flag in next block; the no return hinting whether to make the non-inlined as cold code or not.